File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/config/Environment.java
package uk.gov.hmcts.reform.bulkscan.orchestrator.config;

/**
 * This Singleton contains all the environmental items needed in the integration test profile.
 */
public final class Environment {

    private static final String USER_ID = "640";
    public static final String JURISDICTION = "BULKSCAN";
    public static final String CASE_TYPE_BULK_SCAN = "Bulk_Scanned";
    public static final String CASE_TYPE_EXCEPTION_RECORD = "BULKSCAN_ExceptionRecord";
    public static final String CASE_REF = "1539007368674134";

    private static final String CASE_TYPE_URL = String.format(
        "/caseworkers/%s/jurisdictions/%s/case-types/%s",
        USER_ID,
        JURISDICTION,
        CASE_TYPE_BULK_SCAN
    );
    private static final String CASE_URL = String.format(
        "%s/cases/%s",
        CASE_TYPE_URL,
        CASE_REF
    );
    public static final String GET_CASE_URL = "/cases/" + CASE_REF;
    public static final String CASE_EVENT_URL = CASE_URL + "/events";
    public static final String CASE_SUBMIT_URL = CASE_TYPE_URL + "/cases";
    public static final String CASE_EVENT_TRIGGER_START_URL = CASE_TYPE_URL + "/event-triggers/createException/token";

    private Environment() {
        // utility class construct
    }
}

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/config/Environment.kt
package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers.config

/**
 * This Singleton contains all the environmental items needed in the integration test profile.
 */
object Environment {

    val USER_ID = "640"
    val JURIDICTION = "BULKSCAN"
    const val CASE_TYPE_BULK_SCAN = "Bulk_Scanned"
    const val CASE_TYPE_EXCEPTION_RECORD = "ExceptionRecord"
    val CASE_REF = "1539007368674134"

    private val caseTypeUrl = "/caseworkers/$USER_ID/jurisdictions/$JURIDICTION/case-types/$CASE_TYPE_BULK_SCAN"
    val caseUrl = "$caseTypeUrl/cases/$CASE_REF"
    val caseEventUrl = "$caseUrl/events"
    val caseSubmitUrl = "$caseTypeUrl/cases"
    val caseEventTriggerStartUrl = "$caseTypeUrl/event-triggers/createException/token"
}
